### PR TITLE
Update base Func class and tests

### PR DESCRIPTION
**Pull request by dreadatour**
Remove some operators not supported by all DBs by default. Implement bitwise functions. Improve tests.

Related to https://github.com/iterative/datachain/issues/606 (implement `XOR` function).

### CODE REVIEW

File: src/datachain/func/func.py [hunk 1]
```diff
@@ -96,94 +98,136 @@ def _db_col_type(self, signals_schema: "SignalSchema") -> Optional["DataType"]:
...
+            return Func("div", lambda a: _truediv(a, other), [self], result_type=float)
+        return Func(
+            "div", lambda a1, a2: _truediv(a1, a2), [self, other], result_type=float
+        )
 
     def __rtruediv__(self, other: Union[ColT, float]) -> "Func":
-        return math_truediv(other, self)
+        if isinstance(other, (int, float)):
+            return Func("div", lambda a: _truediv(other, a), [self], result_type=float)
+        return Func(
```
Comment (shcheklein) [inline src/datachain/func/func.py:140]: do we need a test for this?

File: src/datachain/func/func.py [hunk 2]
```diff
@@ -96,94 +98,136 @@ def _db_col_type(self, signals_schema: "SignalSchema") -> Optional["DataType"]:
...
+            "floordiv", lambda a1, a2: _floordiv(a1, a2), [self, other], result_type=int
+        )
 
     def __rfloordiv__(self, other: Union[ColT, float]) -> "Func":
-        return math_floordiv(other, self)
+        if isinstance(other, (int, float)):
+            return Func(
+                "floordiv", lambda a: _floordiv(other, a), [self], result_type=int
+            )
+        return Func(
```
Comment (shcheklein) [inline src/datachain/func/func.py:158]: same here, do we need a test?

**Review by shcheklein — APPROVED** [review]
(please check is need a few more tests)

File: src/datachain/func/func.py (hunk 1 above)
Comment (dreadatour) [reply src/datachain/func/func.py:140]: Here we have it for all three cases: https://github.com/iterative/datachain/pull/641/files#diff-1a98b3b24a2c496f079d2c5f5757a55850ca5840a7e53350e39c5d1c10175979R159-R167

File: src/datachain/func/func.py (hunk 2 above)
Comment (dreadatour) [reply src/datachain/func/func.py:158]: Same here: https://github.com/iterative/datachain/pull/641/files#diff-1a98b3b24a2c496f079d2c5f5757a55850ca5840a7e53350e39c5d1c10175979R189-R197 🤗

**Pull request merged**